cdap-app-fabric/src/main/java/co/cask/cdap/app/runtime/ProgramController.java

```diff
@@ -100,6 +100,20 @@ private State(ProgramRunStatus runStatus) {
       this.runStatus = runStatus;
     }
 
+    /**
+     * Gives the {@link ProgramController.State} for a given {@link ProgramRunStatus}
+     * @param status : the status
+     * @return the state for the status or null if there is no defined state for the given status
+     */
+    public static State getControllerStateByStatus (ProgramRunStatus status) {
```

Comment:
Why do you need this method during upgrade? Shouldn't the upgrade just copy the existing values into different rows?